modules/Sources/Dependencies/ZcashSDKEnvironment/ZcashSDKEnvironmentLiveKey.swift

```diff
@@ -12,7 +12,26 @@ extension ZcashSDKEnvironment: DependencyKey {
     public static let liveValue = Self(
         latestCheckpoint: { network in BlockHeight.ofLatestCheckpoint(network: network) },
         endpoint: { network in
-            LightWalletEndpoint(
+            // In case of mainnet network we may have stored server as a user action in advanced settings
+            if network.networkType == .mainnet {
+                @Dependency(\.userDefaults) var userDefaults
+                
+                let udKey = ZcashSDKEnvironment.Servers.Constants.udServerKey
+                
+                if let storedServerRaw = userDefaults.objectForKey(udKey) as? String,
+                    let storedServer = ZcashSDKEnvironment.Servers(rawValue: storedServerRaw) {
+                    if let endpoint = storedServer.lightWalletEndpoint(userDefaults) {
+                        // Some endpoint is set by a user so we initialize the SDK with this one
+                        return endpoint
+                    } else {
+                        // Endpoint failed, fallback to hardcoded mainnet
```

Comment:
To clarify, AFAICT this does not occur when the _endpoint_ fails, but when _looking up a configured custom endpoint in the key-value store_ fails?